sim/core/item_effects.go

```diff
@@ -10,13 +11,22 @@ import (
 // but there are occasionally class-specific item effects.
 type ApplyEffect func(Agent)
 
+// Function for applying permenent effects to an agent's weapon
+type ApplyWeaponEffect func(Agent, proto.ItemSlot)
```

Comment:
I think this should take a boolean parameter isMH instead of an item slot. Ranged weapon enchants have no reason to use this since they only have 1 slot.

Reply:
The weapon enchant code is extremely verbose because of this. There's no reason that a ranged weapon enchant needs a switch and a default case when a single line a code will do. Same with the weapon enchants, much harder to read because it's not obvious that 'slot' can only actually ever be MH or OH.

Reply:
It's verbose because it (probably) needlessly checks said slot. I've just removed the check.

Reply:
I'm still not crazy about this, but not worth blocking this PR on.